### DISCUSSION TITLE
Does Copilot collect the suggestions?

### DISCUSSION THREAD

**Discussion by prane-eth** [community/community#19328] [Copilot Conversations] [answered]
[Solved]
Does Copilot collect the suggestions which I choose to accept or reject?
I turned off the option to _Allow GitHub to use my code snippets for product improvements_.

In [this page](https://docs.github.com/en/site-policy/github-terms/github-terms-for-additional-products-and-features#github-copilot), it is mentioned "**GitHub does not claim any rights in Suggestions, and you retain ownership of and responsibility for Your Code, including Suggestions you include in Your Code.**"
In [this page](https://github.com/features/copilot#will-my-private-code-be-shared-with-other-users), it is mentioned **"We use data, including information about which suggestions users accept or reject, to improve the model."**

Does it collect the suggestions?

**Comment by davecheney** [comment, marked as answer]
If you have disabled _Allow GitHub to use my code snippets for product improvements_ we will not record the details of the prompt (the source code that was sent to Copilot to complete) or the response(s) from the model.

**Reply by raivatshah** [reply]
Thanks for the answer @davecheney. But say I use copilot (on an individual plan) on a private repo and disable the feature to allow github to use my code for product improvements, will Copilot have access to: 

* secret files (stored in a json for eg) and will it send it back to your servers? 
* all of code and use it, in any way, for product improvements or for any other purpose?

**Reply by davecheney** [reply]
Please see the FAQ for what context is sent to the server as part of the completion request.

If you have disabled _Allow GitHub to use my code snippets for product improvements_ those details will not be stored.
